### PR TITLE
KAA-1241 C SDK QNX guide page

### PR DESCRIPTION
**Pull request by forGGe**

### CODE REVIEW

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md [hunk 1]
```diff
@@ -5,4 +5,81 @@ permalink: /:path/
...
+
+        export QNX_HOST="$QNX_SDK_HOME/host/linux/x86"
+        export QNX_TARGET="$QNX_SDK_HOME/target/qnx6"
+        export PATH="$QNX_HOST/usr/bin:$PATH"
+
+# Creating applications based on C SDK
+
+Since QNX is a POSIX-compliant system, [the Linux guide]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/#c-sdk-build) can be followed to create applications based on C SDK.
+During CMake configuration and building step, make sure [proper CMake toolchain is used](https://cmake.org/cmake/help/v3.0/manual/cmake-toolchains.7.html).
+Kaa C SDK already provides as shown below.
```
Comment (balldir) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: > Kaa C SDK already provides as shown below

Already provides what ? It's confusing.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md [hunk 2]
```diff
@@ -5,4 +5,81 @@ permalink: /:path/
...
+        mkdir -p build
+        cd build
+        cmake -DCMAKE_TOOLCHAIN_FILE=../toolchains/qnx.cmake
+        make
+
+# Exporting application
+
+In order to upload Kaa application to the QNX-running device SSH can be used.
+
+        scp build/<app_name> <user>@<ip_of_target_machine>:<app_name>
```
Comment (balldir) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: SSH is not always available on the QNX systems.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md [hunk 3]
```diff
@@ -5,4 +5,81 @@ permalink: /:path/
...
+# Installing QNX Software Development Platform
+
+To install QNX Software Development Platform (SDP), proceed as follows:
+
+1. [Register](https://www.qnx.com/account/login.html) as a developer and download the following components:
+    - [QNX Software Development Platform 6.6](http://www.qnx.com/download/feature.html?programid=26114)
+    - [QNX Software Development Platform 6.6 Applypatch Patch [Patch ID 4024]](http://www.qnx.com/download/feature.html?programid=26817)
+    - [QNX Software Development Platform 6.6 Header Files Patch [Patch ID 3851]](http://www.qnx.com/download/feature.html?programid=26447)
+
+1. Install [SDP](http://www.qnx.com/developers/articles/inst_5847_9.html). Note: It is recommended to install SDP in the default directory (for Linux platforms, it is `/opt/qnx660`).
```
Comment (balldir) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md:31]: SDP is not a commonly used abbreviation.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md (hunk 1 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: Fixed

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md (hunk 2 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: Added notion of missing SSH

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md (hunk 3 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md:31]: Added notion of the SDP above that

**Comment by balldir** [comment]
LGTM

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md [hunk 4]
```diff
@@ -5,4 +5,83 @@ permalink: /:path/
...
+**NOTE:** The further instructions are expected to be executed on the host machine.
+
+Verified against:
+
+- **Host OS:** **TODO:** [KAA-1241](http://jira.kaaproject.org/browse/KAA-1241) retest this guide against Ubuntu 14.04 LTS 64-bit and Kaa C SDK v.0.10.0.
+- **Target OS:** QNX Neutrino RTOS 6.6.
+
+# Installing QNX Software Development Platform
+
+To install QNX Software Development Platform (SDP hereinafter), proceed as follows:
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: code style

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md [hunk 5]
```diff
@@ -5,4 +5,83 @@ permalink: /:path/
...
+
+# Creating applications based on C SDK
+
+Since QNX is a POSIX-compliant system, [the Linux guide]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/#c-sdk-build) can be followed to create applications based on C SDK.
+During CMake configuration and building step, make sure [proper CMake toolchain is used](https://cmake.org/cmake/help/v3.0/manual/cmake-toolchains.7.html).
+Kaa C SDK already provides the toolchain file for QNX as shown below.
+
+        mkdir -p build
+        cd build
+        cmake -DCMAKE_TOOLCHAIN_FILE=../toolchains/qnx.cmake
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: `cmake -DCMAKE_TOOLCHAIN_FILE=../toolchains/qnx.cmake ..` (missed `..`)

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md [hunk 6]
```diff
@@ -5,4 +5,83 @@ permalink: /:path/
...
+        make
+
+# Exporting application (if SSH is enabled)
+
+If present, you can use SSH on a QNX-running device to transfer the application.
+
+        scp build/<app_name> <user>@<ip_of_target_machine>:<app_name>
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: you're in `build` directory already

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md [hunk 7]
```diff
@@ -5,4 +5,83 @@ permalink: /:path/
...
+
+
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: trailing empty lines

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md [hunk 8]
```diff
@@ -5,4 +5,83 @@ permalink: /:path/
 sort_idx: 90
 ---
 
-SDK integration instructions for select platforms
+* TOC
+{:toc}
+
+{% assign root_url = page.url | split: '/' %}
```
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: `include variables` or something like that.

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md (hunk 5 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: Done

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md (hunk 4 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: Fixed

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md (hunk 6 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: Fixed

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md (hunk 7 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: Deleted

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md (hunk 8 above)
Comment (forGGe) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-QNX-Neutrino/index.md]: Done

**Pull request merged**